packages/dynamic_layouts/lib/src/render_dynamic_grid.dart

```diff
@@ -0,0 +1,402 @@
+            parentUsesSize: true,
+          );
+        }
+        trailingChildWithLayout = child;
+      }
+      assert(child != null);
+      final SliverGridParentData childParentData =
+          child!.parentData! as SliverGridParentData;
+      gridGeometry = layout.updateGeometryForChildIndex(index, child!.size);
+      childParentData.layoutOffset = gridGeometry.scrollOffset;
+      childParentData.crossAxisOffset = gridGeometry.crossAxisOffset;
+      assert(childParentData.index == index);
+      // The current child may not extend past a previously laid out child.
+      endScrollOffset = math.max(
+        endScrollOffset,
+        childScrollOffset(child!)! + paintExtentOf(child!),
+      );
+      return true;
+    }
+
+    // Find the first child that ends after the scroll offset.
+    while (endScrollOffset < scrollOffset) {
+      leadingGarbage += 1;
+      if (!advance()) {
+        assert(leadingGarbage == childCount);
+        assert(child == null);
+        // we want to make sure we keep the last child around so we know the end
+        // scroll offset
+        collectGarbage(leadingGarbage - 1, 0);
+        assert(firstChild == lastChild);
+        final double extent =
+            childScrollOffset(lastChild!)! + paintExtentOf(lastChild!);
+        geometry = SliverGeometry(
+          scrollExtent: extent,
+          maxPaintExtent: extent,
+        );
+        return;
+      }
+    }
+
+    // Now find the first child that ends after our end.
+    while (endScrollOffset < targetEndScrollOffset ||
+        !layout.reachedTargetScrollOffset(targetEndScrollOffset)) {
+      if (!advance()) {
+        reachedEnd = true;
+        break;
+      }
+    }
+
+    // Finally count up all the remaining children and label them as garbage.
+    if (child != null) {
+      child = childAfter(child!);
+      while (child != null) {
+        trailingGarbage += 1;
+        child = childAfter(child!);
+      }
+    }
+
+    // At this point everything should be good to go, we just have to clean up
+    // the garbage and report the geometry.
+    // We only collect trailing garbage because
+    //
+    // 1 - dynamic tile placement is dependent on the preceding tile,
+    //     potentially, the SliverGridLayout could cache the placement of each
+    //     tile to refer back to, but that would mean tiles could not change in
+    //     size or be added/removed without having to recompute the whole layout.
+    //
+    // 2 - Currently, Flutter only supports collecting garbage in sequential
+    //     order. Dynamic layout patterns can break this assumption. In order to
+    //     truly collect garbage efficiently, support for nonsequential garbage
+    //     collection is necessary.
+    // TODO(all): document this behavior so users are aware of this temporary
+    //   also - file a bug in https://github.com/flutter/flutter/labels/p%3A%20dynamic_layouts
+    // trade off
```

Comment:
The "trade off" appears lonely? Should this be here?

Reply:
Oh, I think this belong to the end of line 353. Basically, 354 needs to move down one line.